backend/inspirehep/factory.py

```diff
@@ -22,6 +23,29 @@ def config_loader(app, **kwargs_config):
     return result
 
 
+class DispatcherMiddlewareWithoutHostNameChecking(DispatcherMiddleware):
+    """WSGI middleware without checking hostname.
+
+    It will override the request ``HOST`` to always match
+    the ``SERVER_NAME``.
+    """
+
+    def __call__(self, environ, start_response):
+        if "SERVER_NAME" in environ:
```

Comment:
this is always present according to the spec: https://www.python.org/dev/peps/pep-0333/#environ-variables

Reply:
on the other hand this is passed by the WSGI server, so it's probably not the same thing as `app.config['SERVER_NAME']`